Add Backstage intro page (#2238)
giantswarm/docs: src/content/overview/developer-portal/_index.md

---
title: Developer portal

> description: Our Backstage-based developer portal is your engineer's front end to the platform. We provide our self-service user interface as plugins for Backstage, so engineers using your platform find all the information they need in the right place.

Reviewer: Please avoid our/your/we/you. This description is also too long for readers to take a look, and contains lots of fill words. I recommend extending the previous description a little:

```suggestion
description: The Giant Swarm platform's self-service portal for platform engineers and developers. Based on Backstage.
```

weight: 30
menu:
  principal:
    parent: overview
    identifier: overview-developer-portal
last_review_date: 2024-07-01
owner:
  - https://github.com/orgs/giantswarm/teams/team-honeybadger
---

> Our developer portal absed on [Backstage](https://www.cncf.io/projects/backstage/) is your engineer's front end to the platform. We provide our self-service user interface as plugins for Backstage, so that engineers using your platform find all the information they need in the right place.

Reviewer: Reworded a lot. I don't see how "Backstage plugin" relates to finding stuff in the right place (and that last part of the sentence can be left off without losing information, i.e. "fill words" in my review-nit lingo, because a frontend always tries to put stuff in some right place).

```suggestion
The Giant Swarm platform's developer portal, based on [Backstage](https://www.cncf.io/projects/backstage/), is a frontend for customer's engineers to work with the platform. The self-service user interface is provided as plugins for Backstage.
```

Author: I clarified the sentence. The idea is that by integrating our UI in the portal, devs find the information we provide in the place they visit often anyway.

> Giant Swarm can assist you in various ways depending on where you are in your journey towards your own developer portal. We provide a pre-configured Backstage instance as a starting point to customers who don't run their own instance yet. We also provide plugins that can be used to extend the functionality of your Backstage instance.

Reviewer: ```suggestion
Customers can either consume this pre-configured Backstage instance if they do not have their own developer portal yet. Or the Giant Swarm plugins for Backstage can be integrated into an existing customer Backstage installation in order to gain the same functionality, for example to manage clusters on the Giant Swarm platform.
```

> ## Why developer portals

Reviewer: Can we reword this? It sounds like an incomplete sentence where a verb should follow.

P.S. "Papa, kann ich Schokolade?" – "'...wegschmeißen' oder was?"

Reviewer 2: What about leaving the content under the intro removing this header?

> The Giant Swarm platform combines capabilities provided by a variety of open source tools, and likely your engineers are using even more tools to accomplish their goals. Developer portals should serve as a single entry point and provide a navigation path to the tools engineers need access to. Furthermore, they can give an overview of the software landscape and the platform itself, make documentation and troubleshooting runbooks accessible, and assist in the creation of new services according to platform standards.

Reviewer: This doesn't say much about which capabilities. Can we link to the platform overview, or give examples here, without making this too long? Developer portals are a topic where many companies don't see the benefit, so examples are critical.

And let's add `runbooks` to the allowlist of vale.

Author: > Developer portals are a topic where many companies don't see the benefit, so examples are critical.

I don't second this. Also, we'll add more details once we can.

## Why Backstage

> Backstage, originally developed by Spotify and donated to the Cloud Native Computing Foundation in 2020, is an open-source framework for building developer portals. With more than 1.500 contributors and a fast-paced release cycle, Backstage is improving and extending its capabilities rapidly. More and more enterprises are adopting Backstage as their foundation for developer portals, creating [plugins](https://backstage.io/plugins/) and open-sourcing them to the community.

Reviewer: ```suggestion
Backstage, originally developed by Spotify and donated to the Cloud Native Computing Foundation in 2020, is an open-source framework for building developer portals. With more than 1500 contributors and a fast-paced release cycle, Backstage is improving and extending its capabilities rapidly. More and more enterprises are adopting Backstage as their foundation for developer portals, creating [plugins](https://backstage.io/plugins/) and open-sourcing them to the community.
```

(`1.500` in English is one comma five)

> ## Core features

Reviewer 2: ```suggestion
## Capabilities
```
To align with the rest of the overview pages

> Backstage as we provide it to customers and as we run it internally at Giant Swarm comes with a set of core features:

Reviewer: ```suggestion
The pre-configured Backstage instance provides the following functionality:
```

(again avoiding we/you/repetition)

- **Cluster overview**: An overview of all Kubernetes clusters managed through the Giant Swarm platform.
- **Catalog**: A catalog of all applications that Giant Swarm publishes in app catalogs. As a customer, you can extend this catalog with your own applications, services, and more.
- **Deployment details**: Where applicable, you get access to deployment information, showing the status of the respective service's deployment in your clusters.

> - **Templates**: Fast and easy creation of new clusters, services, and more through templates.

Reviewer: We best add screenshots to showcase the benefits. Otherwise it's hard to grasp what users get ("What's in it for me?").

Author: Yes, but this is only an intro page and we are not ready for features yet.
